src/main/java/com/google/devtools/build/lib/query2/engine/Lexer.java

```diff
@@ -259,7 +260,12 @@ private String scanWord(char firstChar) {
           }
         }
         default -> {
-          return bufferSlice(oldPos, pos);
+          // All remaining ASCII characters (and only ASCII characters) are query operators.
+          if (c <= 0x7F) {
+            return bufferSlice(oldPos, pos);
+          }
+          // Unicode characters are allowed in words, so we continue scanning.
+          pos++;
```

Comment:
This isn't fixing a regression, but it makes good on the "8.x brings reliable Unicode support" promise as the error message shown before this commit was inscrutable (it contained individual Unicode bytes turned into latin-1).